docs/documentation/siddhi-4.0.md

```diff
@@ -104,63 +101,66 @@ The source syntax is as follows:
-* <a target="_blank" href="https://wso2-extensions.github.io/siddhi-io-rabbitmq/">RabbitMQ</a> 
-* In-Memory
-* <a target="_blank" href="https://wso2-extensions.github.io/siddhi-io-file/">File</a> _(Only works in WSO2 Stream Processor)_ 
-* <a target="_blank" href="https://wso2-extensions.github.io/siddhi-io-jms/">JMS</a> _(Only works in WSO2 Stream Processor)_  
-* <a target="_blank" href="https://wso2-extensions.github.io/siddhi-io-email/">E-mail</a>  _(Only works in WSO2 Stream Processor)_ 
-* <a target="_blank" href="https://wso2-extensions.github.io/siddhi-io-email/">WSO2-Event</a> _(Only works in WSO2 Stream Processor)_ 
+depends on the source type selected, some of the the parameters are optional. </br>
+
+For detailed information about the parameters see the documentation for the relevant source.
+
+The following is the list of source types that are currently supported:
+
+* [HTTP](https://wso2-extensions.github.io/siddhi-io-http/) _(Only works in WSO2 Stream Processor)_
+* [Kafka](https://wso2-extensions.github.io/siddhi-io-kafka/)
+* [TCP](https://wso2-extensions.github.io/siddhi-io-tcp/)
+* In-memory
+* [WSO2Event](https://wso2-extensions.github.io/siddhi-io-wso2event/)
+* [Email](https://wso2-extensions.github.io/siddhi-io-email/) _(Only works in WSO2 Stream Processor)_
+* [JMS](https://wso2-extensions.github.io/siddhi-io-jms/) _(Only works in WSO2 Stream Processor)_
+* [File](https://wso2-extensions.github.io/siddhi-io-file/) _(Only works in WSO2 Stream Processor)_
+* [RabbitMQ](https://wso2-extensions.github.io/siddhi-io-rabbitmq/)
+* [MQTT](https://wso2-extensions.github.io/siddhi-io-mqtt/)
 
 **Source Mapper**
 
-Each `@source` will have a mapping denoted by `@map` that converts the incoming message format to Siddhi event.
+Each `@source` configuration has a mapping denoted by the `@map` annotation that converts the incoming messages format to Siddhi events.
 
-The `type` parameter of `@map` defines the map type that's responsible of mapping the data. The other parameters to be 
-configured depends on the mapper selected, some of those can also be optional values. 
-For detailed information about the parameters refer the appropriate mapper documentation.
+The `type` parameter of the `@map` defines the map type to be used to map the data. The other parameters to be 
+configured depends on the mapper selected. Some of these parameters are optional. </br>
+For detailed information about the parameters see the documentation for the relevant mapper.
 
 !!! tip 
-    When `@map` is not provided `@map(type='passThrough')` will be used as default. This can be used when source consumes Siddhi events and when it does not need any mappings.
+    When the `@map` annotation is not provided, `@map(type='passThrough')` is used as default. This default mapper type can be used when source consumes Siddhi events and when it does not need any mappings.
     
+The following is a list of currently supported source mapping types:
```

Comment:
This has to have after explaining about map attributes because that's also part of map

Reply:
Corrected